### PR TITLE
[Documentation/platforms/xtensa/esp32s3/boards|boards/xtensa/esp32s3]Document and Defconfig for Esp32s3 Sdmmc Support

### PR DESCRIPTION
**Pull request by Windrow14**
## Summary

Add documentations and default configurations for sdmmc support feature on esp32s3.
See details at https://github.com/apache/nuttx/pull/12898#issuecomment-2296478840.

## Impact

Documentation/platforms/xtensa/esp32s3/boards/esp32s3-devkit: description for esp32s3-devkit with external adapter and emmc module.
Documentation/platforms/xtensa/esp32s3/boards/esp32s3-korvo-2: description for esp32s3-korvo-2 with SD slot enabled.
boards/xtensa/esp32s3/esp32s3-devkit/configs/sdmmc/defconfig: defconfig for esp32s3-devkit with external adapter and emmc module.
boards/xtensa/esp32s3/esp32s3-korvo-2/configs/sdmmc/defconfig: defconfig for esp32s3-korvo-2 with SD slot enabled.

## Testing

See in https://github.com/apache/nuttx/pull/12898.

### CODE REVIEW

**Comment by Windrow14** [comment]
@JackyCaoYJ

**Comment by Windrow14** [comment]
Hi @tmedicci,
Sorry to bother you. Could you please verify the default configuration on esp32s3-korvo-2 in your spare time? Many thanks!

**Comment by tmedicci** [comment]
> Hi @tmedicci, Sorry to bother you. Could you please verify the default configuration on esp32s3-korvo-2 in your spare time? Many thanks!

Of course, I will check right now.

**Review by tmedicci — COMMENTED** [review]
I was able to test it on ESP32-S3-Korvo-2 board but it worked only under specific circumstances:

First of all, I had to enable `CONFIG_FS_FAT` in menuconfig. After building the firmware, it didn't work as expected. It fails with the SD card in the slot.
```
nsh> mount -t vfat /dev/mmcsd1 /mnt
nsh: mount: mount failed: 19
```

Then, I tried to enable `CONFIG_DEBUG_FEATURES`, `CONFIG_DEBUG_ASSERTIONS` and `CONFIG_DEBUG_MEMCARD_[ERROR|WARN|INFO]`. Initially, it was asserting at 
`esp32s3_sdmmc.c:524`, then I removed [this](https://github.com/apache/nuttx/blob/190c8787ff7e38e87adac6e8dc4a7696c3fbd0e3/arch/xtensa/src/esp32s3/esp32s3_sdmmc.c#L524) line (please note that it isn't required because that situation is checked in the following lines).

Finally, I was able to get more information and by running `mkfatfs -F 32 -r /mnt /dev/mmcsd1` and `mount -t vfat /dev/mmcsd1 /mnt` it worked: but this is not consistent... if I reboot the board and try to mount once again it may fail with other errors (and eventually it works):
```
NuttShell (NSH) NuttX-10.4.0
nsh> mount -t vfat /dev/mmcsd1 /mnt
fat_checkbootrecord: WARNING: Signature: 2fbc FS sectorsize: 17819 HW sectorsize: 512
esp32s3_interrupt: ERROR: Data timeout, pending=00000200 remaining: 512
mmcsd_eventwait: ERROR: Awakened with 0c
mmcsd_readsingle: ERROR: CMD17 transfer failed: -110
fat_mount: ERROR: Failed to read sector -796919980: -110
fat_mount: ERROR: No valid boot record
nx_mount: ERROR: Bind method failed: -22
nsh: mount: mount failed: 22
nsh> mount -t vfat /dev/mmcsd1 /mnt
mmcsd_recv_r1: ERROR: R1=80000900
mmcsd_readsingle: ERROR: mmcsd_recv_r1 for CMD17 failed: -5
nx_mount: ERROR: Bind method failed: -5
nsh: mount: mount failed: 5
nsh> mount -t vfat /dev/mmcsd1 /mnt
```

In summary: without `CONFIG_DEBUG_FEATURES` it never works (returns -19), with debug features enabled, it sometimes works. May this be a timing issue? Can you please test with ESP32-S3-DevKitC board too (with the external slot)?

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by Windrow14** [comment]
@tmedicci Thank you very much for your help!
We've spent some time on investigating the problem you've encountered on ESP32S3-Korvo-2. We've got an external SD slot so that we can do tests with SD slot on ESP32S3-DevKit now. We can reproduce the phenomenon you've described on it.

1. About the assertion you've mentioned, it is triggered due to a non-zero return in `mmcsd_cardidentify`, which is expected for SD cards but does not happen for MMCs. In `esp32s3_waitresponse`, two events happen in this case. One event is handled, and one is received after returning, this cause problem to later command. So, when debugging is enabled, more time is delayed, the later command will not be effected.
2. Other unstable problems are due to a read problem. We've found that for the first several times of reading sector from SD card, the beginning of the content is not correct. The more readings, the less incorrect content length. Once all content is correct for the first time, then the reading problem is gone. Still investigating for the root cause.
3. We do not have to use FAT file system, I've also tried littlefs on DevKit. Thus, I did not add `CONFIG_FS_FAT`in the default config. That is a good idea to describe how to format and mount the sd/mmc device.

**Comment by Windrow14** [comment]
>     1. About the assertion you've mentioned, it is triggered due to a non-zero return in `mmcsd_cardidentify`, which is expected for SD cards but does not happen for MMCs. In `esp32s3_waitresponse`, two events happen in this case. One event is handled, and one is received after returning, this cause problem to later command. So, when debugging is enabled, more time is delayed, the later command will not be effected.
> 
>     2. Other unstable problems are due to a read problem. We've found that for the first several times of reading sector from SD card, the beginning of the content is not correct. The more readings, the less incorrect content length. Once all content is correct for the first time, then the reading problem is gone. Still investigating for the root cause.
> 
>     3. We do not have to use FAT file system, I've also tried littlefs on DevKit. Thus, I did not add `CONFIG_FS_FAT` in the default config. That is a good idea to describe how to format and mount the sd/mmc device.

Fix to 1: returning after CMDDONE is handled, check details at https://github.com/apache/nuttx/pull/13165.
Fix to 2: use SDIO_DMARECVSETUP instead of SDIO_RECVSETUP, check details at https://github.com/apache/nuttx/pull/13164.
Fix to 3: already merged in this PR.

**Comment by Windrow14** [comment]
Update:

> Fix to 2: use SDIO_DMARECVSETUP instead of SDIO_RECVSETUP, check details at https://github.com/apache/nuttx/pull/13164.

This PR has not been merged so far, please apply this patch for personal use for the moment: https://github.com/apache/nuttx/commit/5e64f007c4050d001a4db60380186df28b710114.
